climada_petals/entity/exposures/osm_dataloader.py

```diff
@@ -0,0 +1,346 @@
+
+        return gdf_nodes
+
+    def _update_availability(self, full_set, to_remove):
+        """
+        update id availabilities from whole result set after using up some for
+        geometry construction
+        """
+        return [item for item in full_set if item not in to_remove]
+
+    def _assemble_results(self, result, closed_lines_are_polys=True):
+        """
+        assemble an overpass result object with results, ways, nodes etc. from the
+        format-specific structure into one geodataframe
+
+        Parameters
+        ---------
+        result : overpy.Overpass result object
+        closed_lines_are_polys : bool
+                whether closed lines are polygons. Default is True
+
+        Returns
+        ------
+        gdf_results : gpd.GeoDataFrame
+            Result-gdf from the overpass query.
+        """
+        gdf_results = gdf_ways = gdf_nodes = gpd.GeoDataFrame(
+            columns=['osm_id','geometry','tags'])
+        nodes_avail = result.node_ids
+        ways_avail = result.way_ids
+
+        if len(result.relations) > 0:
+            nodes_taken, ways_taken, gdf_rels = self._assemble_from_relations(result)
+            gdf_results = gdf_results.append(gdf_rels)
+            nodes_avail = self._update_availability(nodes_avail, nodes_taken)
+            ways_avail = self._update_availability(ways_avail, ways_taken)
+
+        if len(ways_avail) > 0:
+            nodes_taken, gdf_ways = self._assemble_from_ways(
+                result,  ways_avail, closed_lines_are_polys)
+            gdf_results = gdf_results.append(gdf_ways)
+            nodes_avail = self._update_availability(nodes_avail, nodes_taken)
+
+        if len(nodes_avail) > 0:
+            gdf_nodes = self._assemble_from_nodes(result, nodes_avail)
+            gdf_results =  gdf_results.append(gdf_nodes)
+
+        if len(result.nodes) == 0:
+            LOGGER.warning('empty result gdf returned.')
+
+        return gdf_results.reset_index(drop=True)
+
+    def _osm_geoms_to_gis(self, gdf):
+        """
+        convert lat / lon ordering of OSM convention back to conventional
+        GIS ordering (x,y) instead of (lat / lon)
+
+        Parameters
+        ----------
+        gdf : gpd.GeoDataFrame
+
+        Returns
+        -------
+        gpd.GeoSeries
+            Geometry series with swapped coordinates.
+        """
+
+        return gdf.geometry.map(lambda geometry:
+                                shapely.ops.transform(
+                                    lambda x, y: (y, x), geometry))
+
+    def get_data_overpass(self, closed_lines_are_polys=True):
+        """
+        wrapper for all helper funcs to get & assemble data
+        """
+
+        query_clause = self._overpass_query_string()
+        result = self._insistent_osm_api_query(query_clause)
+        gdf_result = self._assemble_results(result, closed_lines_are_polys)
+        gdf_result['geometry'] = self._osm_geoms_to_gis(gdf_result)
```

Comment:
Geometry should be set explicitly, see https://geopandas.org/en/stable/docs/reference/api/geopandas.GeoDataFrame.set_geometry.html
```suggestion
        gdf_result = gdf_result.set_geometry(self._osm_geoms_to_gis(gdf_result))
```